samcli/lib/providers/sam_base_provider.py

```diff
@@ -61,14 +68,28 @@ def _extract_lambda_function_code(resource_properties: Dict, code_property_key:
         str
             Representing the local code path
         """
-
         codeuri = resource_properties.get(code_property_key, SamBaseProvider.DEFAULT_CODEURI)
 
         if isinstance(codeuri, dict):
             return SamBaseProvider.DEFAULT_CODEURI
 
         return cast(str, codeuri)
 
+    @staticmethod
+    def _is_code_uri_s3(codeuri: Optional[Any]) -> bool:
```

Comment:
nit: isn't `Optional[Any] == Any`?

Reply:
Can codeuri be anything other than None, dict, str? Shouldn't we use something like Optional[str, Dict, None]?

Reply:
> nit: isn't Optional[Any] == Any?
yeah right.

I changed to `Optional[Union[str, Dict]]`